### ISSUE TITLE
PyOMP and Numba

### ISSUE THREAD

**Issue by raviapatel** [numba/numba#9002]

**Comment by esc** [comment]
> ## Feature request
> Dear Numba team, Thank you for the wonderful tool. I found PyOMP (https://github.com/Python-for-HPC/PyOMP) tool which is quite interesting as it enhances multithreading implementation of numba. Neverthless I cannot find in documentation if this project was merged with numba offical implementation. Is there plan to implement such a approach in numba ?

This was not merged and is unlikely to be merged. Instead it will remain an extension project, I believe.

@DrTodd13 can you comment on this?

**Comment by DrTodd13** [comment]
Yes, at this point, it seems unlikely to be merged into mainline Numba but more thoughts below.

I suppose there are two kinds of extension projects.  One in which the whole code base is duplicated/modified and one in which the new functionality is more like a patch on top of an unmodified existing project.  Currently, PyOMP is the former category but we are moving in the direction of the latter.  That isn't that visible to the end-user but will allow us to keep up-to-date with the latest core Numba easier.

I have mentioned this issue in the Numba meetings but if Numba wants to be a compiler toolkit then there is going to be this issue of people finding out about the Numba extensions/variants and installing them and even using them in combination.  I've started suggesting an approach where Numba isn't maintaining the extension code but is providing an infrastructure for automatically loading extensions when certain data types or constructs not supported by core Numba are detected.

The problem with that is that the source of any errors may be hidden and people may report everything to Numba when the error is in an extension.  Perhaps we can do something to work around this issue and reduce the impact on core Numba devs.

**Comment by raviapatel** [comment]
@DrTodd13  thank you for your explaination. In this case question for you will be does th pyOMP support all features of numba i.e. if one uses just pyOMP and not numba?
I mean for instance does cuda functionality would work too or for that one has to switch back to numba?

**Comment by DrTodd13** [comment]
@raviapatel Everything in Numba including cuda should still work.
